### PR TITLE
add roles and jwt auth

### PR DESCRIPTION
**Pull request by phyliciachristel**
- added roles (user, moderator, admin)
- allow admin to change roles between user and moderator
- prevent users that are not signed in from accessing questions page
- only allow admin to add or delete a question
- added jwt token and user stored in cookies

Others:
- .env has to be updated to include `JWT_SECRET_KEY`
- have not implemented password encryption or refresh tokens

### CODE REVIEW

File: frontend/src/components/questions/questionInfo.js [hunk 1]
```diff
@@ -23,10 +23,26 @@ const BootstrapDialog = styled(Dialog)(({ theme }) => ({
 
 function QuestionInfo({ open, handleClose, question, handleDelete }) {
   // if there is no selected question, return null
```
Comment (cfyjoanna) [inline frontend/src/components/questions/questionInfo.js:25]: u yeet my comment away

Comment (cfyjoanna) [reply frontend/src/components/questions/questionInfo.js]: thank u

**Comment by cfyjoanna** [comment]
LGTM 👍

**Review by farisanadia — APPROVED** [review]
looks good 2 me les gaur

File: backend/middleware/auth.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    try {
+        const token = req.cookies.token;
+
+        if(token) {
+            const decode = jwt.verify(token, jwtSecretKey);
+            if(decode) {
+                console.log("verified");
+                next();
+            }
+        }else{
```
Comment (farisanadia) [inline backend/middleware/auth.js:15]: space btw } else {

Comment (farisanadia) [reply backend/middleware/auth.js:15]: and after the if statements above

File: frontend/src/components/navBar.js [hunk 3]
```diff
@@ -19,10 +21,15 @@ function NavBar() {
     };
 
     useEffect(() => {
-        const loggedInUser = localStorage.getItem('user');
+        const loggedInUser = Cookies.get('user');
+        setIsAdmin(false);
```
Comment (farisanadia) [inline frontend/src/components/navBar.js:25]: why is this setAdmin needed here?

**Review by farisanadia — COMMENTED** [review]
looks good once the comments are resolved

File: frontend/src/components/navBar.js (hunk 3 above)
Comment (phyliciachristel) [reply frontend/src/components/navBar.js:25]: same as setIsModerator

**Pull request merged**